src/components/ChatEntry.js

```diff
@@ -1,22 +1,49 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './ChatEntry.css';
 import PropTypes from 'prop-types';
 
 const ChatEntry = (props) => {
+  const [liked, setLiked] = useState('');
```

Comment:
This state variable can also be lifted to App. If the chatMessages are stored in a state variable, then it's possible to create a callback that will update the liked value based on the id of the message. That function can be passed down to this component, and then it can be called in onClick function that starts on line 28. This will trigger a re-render at the App level, which will pass down the updated props, and this component will be get an updated version of it's props with the new value for liked.